File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/JavaScript/_017)_Leaders_in_an_Array.js
//if current element/person value is greater than front element/person then current element is considered as a leader
function leaders(arr) {
    let leaders = [];
    for (let i = 0; i < arr.length; i++) {
        let flag = false;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[i] <= arr[j]) {
                flag = true;
                break;
            }
        }
        if (flag == false) leaders.push(arr[i]);
    }
    return leaders;
}

let arr = [7, 10, 4, 3, 6, 5, 2];
console.log(`Leaders = ${leaders(arr)}`);

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_017)_Leaders_in_an_Array.dart
/*
If current element/person value is greater than front element/person 
then current element is considered as a leader
*/

List findLeaders(arr) {
  List leaders = [];
  for (int i = 0; i < arr.length; i++) {
    var flag = false;
    for (var j = i + 1; j < arr.length; j++) {
      if (arr[i] <= arr[j]) {
        flag = true;
        break;
      }
    }
    if (flag == false) {
      leaders.add(arr[i]);
    }
  }
  return leaders;
}

void main() {
  var arr = [7, 10, 4, 3, 6, 5, 2];
  var leaders = findLeaders(arr);
  print('Leaders in the array are: $leaders');
}
